migrations.go

```diff
@@ -22,6 +22,8 @@ func Migrate(store Store) error {
 		return err
 	}
 
+	store.DoPreMigration()
```

Comment:
is there a reason that this migration needs to occur before the other migrations? also, I think that the migration should occur in this file and not be assigned on the store struct - the store shouldn't need to know about what was required to migrate it, I think?

Reply:
Yes, the file moving operation is an actual pre-migration, because the whole migration is operating already according to the new database structure. If the user sets the new shared path, we move the files there, and only after we moved the files we do the actual migration.

We surely could move the files after the migrations occurred. The real answer is that would require waaay more code changes, so I guess this is the most efficient way to do it.

About ``DoPreMigration()`` being in store, I guess that's because it's an actual JSONStore method which operates on its own JSONStore. Also, in this way we don't have to check whether we have a JSON or a boltdb, because the boltdb has its own pre-migration, which does absolutely nothing.

That's the way I see it obviously :P